### PR TITLE
[TIMOB-16369] Android: Disable popup / contextual bar inside webview when text-selecting

### PR DESCRIPTION
**Pull request by frankieandone**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-16369

### CODE REVIEW

**Comment by hansemannn** [comment]
Please rebase with master, thanks!

**Comment by sgtcoolguy** [comment]
@fmerzadyan To rebase and remove the extra conflicting commit:

```
git checkout master
git pull upstream master
git checkout TIMOB-16369
git rebase master
git push origin TIMOB-16369 --force
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 1]
```diff
@@ -28,11 +29,14 @@
 import org.appcelerator.titanium.proxy.TiViewProxy;
 import org.appcelerator.titanium.util.TiConvert;
 import org.appcelerator.titanium.util.TiMimeTypeHelper;
+import org.appcelerator.titanium.util.TiRHelper;
 import org.appcelerator.titanium.util.TiUIHelper;
 import org.appcelerator.titanium.view.TiBackgroundDrawable;
 import org.appcelerator.titanium.view.TiCompositeLayout;
 import org.appcelerator.titanium.view.TiUIView;
 
+//import ti.modules.titanium.ui.R;
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: remove this commented out line

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 2]
```diff
@@ -72,32 +73,113 @@
...
+	private class TiWebView extends WebView {
 		public TiWebViewClient client;
+		protected ActionMode actionMode = null;
+		protected ActionMode.Callback actionModeCallback = null;
 
-		public TiWebView(Context context)
-		{
+		public TiWebView(Context context) {
 			super(context);
+//            Log.d(TAG, "TiUIWebView.TiWebView constructor is fired");
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: remove commented out logging

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 3]
```diff
@@ -72,32 +73,113 @@
...
 	}
-	
+
 	private reloadTypes reloadMethod = reloadTypes.DEFAULT;
 	private Object reloadData = null;
-	
-	private class TiWebView extends WebView
-	{
+
+	private class TiWebView extends WebView {
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: Looks like you're not using the same formatting preferences we had, so there are a lot of formatting changes here.

Please use the formatting settings from our standards repo: https://github.com/appcelerator/appc-standards/tree/master/languages/java

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 4]
```diff
@@ -72,32 +73,113 @@
...
 			super(context);
+//            Log.d(TAG, "TiUIWebView.TiWebView constructor is fired");
 		}
 
+		//TIMOB-16369 called for custom contextual action bar
 		@Override
-		public void destroy()
-		{
+		public ActionMode startActionMode(ActionMode.Callback callback) {
+//            Log.d(TAG, "TiWebView.startActionMode has been fired");
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: Remove all this commented out code...

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
I'm struggling to understand what exactly we're trying to achieve here. The linked ticket is not very clear and has a long comment history.

Are we trying to expose some new property to disable the standard popup/context bar when used? Or allow the user to create their own popup bar contents?

From what I can tell the user wanted to stop those default options in the action bar, but still allow you to select text (Basically you can select/highlight text, but can't copy/paste it out). From what I can make out of @hansemannn 's iOS implementation is that his new property actually disables both text selection and the popup, which doesn't seem like what the user wanted.

Additionally, it looks like Tim Poulsen was asking to replace the default action bar content with his own content.

I think we need to go back to the drawing board on this ticket and properly address what the user is asking for (if possible), including on iOS.

**Comment by hansemannn** [comment]
Closing in favor of #8494.

**Pull request closed without merging**